ch_2/tuples/named_tuples.py

```diff
@@ -0,0 +1,62 @@
+# Exerice 1: Use namedtuple and write a Python program that takes the people list (below), and produces a table.
+from collections import namedtuple
+
+people = [
+    ("Donald", "Trump", 7.85),
+    ("Vladimir", "Putin", 3.626),
+    ("Jinping", "Xi", 10.603),
+]
+
+
+def table_of_peope(people):
+
+    Person = namedtuple("Person", ["last", "first", "time"])
+
+    update = ""
+    for person in people:
+        update += "{0:10} {1:10} {2:5.2f}\n".format(*Person._make(person))
+    return update.strip()
+
+
+print(table_of_peope(people))
+
+# Exercise 2: Define a list of tuples, in which each tuple contains the name, length (in minutes), and director of the movies nominated for "best picture" Oscar awards last year.
+# Ask the user whether they want to sort the list by title, length, or director’s name,
+# and then present the list sorted by the user’s choice of axis.
+# The user can input 1, 2, or 3 fields to sort by.
+
+import operator
+
+movies = [
+    ("Winter in Boston", 120, "Jill McFee"),
+    ("Adventures of P&A", 147, "Arnold Masters"),
+    ("Move to Florida", 90, "George Bee"),
+]
+
+print("Sort the movies by title, length or director. You can sort one or all fields.")
+print()
+user_sort = input(
+    "Enter the field(s) you want to sort by, with each field separated by a comma: "
+)
+
+
+def movie_dictionary(movies):
+    """Return a dictionary of movie namedtuples"""
+    Movie = namedtuple("Movie", ["title", "length", "director"])
+
+    all_movies = []
+    for movie in movies:
+        each_movie = Movie(*movie)
+        all_movies.append(each_movie._asdict())
```

Comment:
Why can't you just append the named tuple instead of calling `_asdict()`?

Reply:
@paulghaddad No, it appears that I must use _asdict() because if I send a list to the sort_movie_output(user_sort), I get an error.